crates/starknet-types-core/src/short_string/mod.rs

```diff
@@ -42,7 +50,45 @@ impl From<ShortString> for Felt {
     }
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Copy, Clone)]
+pub enum TryShortStringFromFeltError {
+    TooLong,
+    NonAscii,
+}
+
+impl core::fmt::Display for TryShortStringFromFeltError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            TryShortStringFromFeltError::TooLong => "string to long",
+            TryShortStringFromFeltError::NonAscii => "string contains non ascii characters",
+        }
+        .fmt(f)
+    }
+}
+
+impl TryFrom<Felt> for ShortString {
+    type Error = TryShortStringFromFeltError;
+
+    fn try_from(value: Felt) -> Result<Self, Self::Error> {
+        let bytes = value.to_bytes_be();
+        if bytes[0] != 0 {
+            return Err(TryShortStringFromFeltError::TooLong);
+        }
+        let first_non_zero_byte = match bytes.iter().position(|&v| v != 0) {
+            Some(i) => i,
+            None => return Ok(ShortString(String::new())),
+        };
+        if !bytes[first_non_zero_byte..].is_ascii() {
+            return Err(TryShortStringFromFeltError::NonAscii);
+        }
+
+        let s = unsafe { str::from_utf8_unchecked(&bytes[first_non_zero_byte..]) };
```

Comment:
I would add a comment explaining why is it safe to call `from_utf8_unchecked` here (It is because of the condition in lines 81-83, right?)